#167729790 Ft logout user session

davidfrank96 opened the pull request:
#### What does this PR do?
#This is a logout endpoint were users can logout
#### Description of Task to be completed?
#- created a logout.,js file in the controller
#-updated the index file for routes
#-created a logout.route.js file 
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#167729790
#### Screenshots (if appropriate)
#### Questions:
N/A

davidfrank96 committed:
feat(logout)
- created a logout.,js file in the controller
-updated the index file for routes
-created a logout.route.js file
[Finishes #167729790]

davidfrank96 committed:
feat(dotenv)
- importing dotenv file
- removed the redirecting, and added a statuscode with a string object returning 'you have have logged out successfully'
[Finishes #167729790]

davidfrank96 committed:
feat(test)
- implemented tests for the logout endpoint
- use eslint to lint the codebase
-implemented JsDocs
-included an index.js file in the test folder for multiple testing
[Finishes #167729790]

chibytez commented on src/database/migrations/20190818182539-create-user.js:
```diff
@@ -32,5 +32,5 @@ export default {
     }
   }),
-  down: queryInterface => queryInterface.dropTable('Users')
+  down: (queryInterface) => queryInterface.dropTable('Users')
```
Nice work, can we remove the parentheses around queryInterface

davidfrank96 replied:
okay...

chibytez commented on src/database/seeders/20190817154557-User.js:
```diff
@@ -25,5 +25,5 @@ export default {
 
   ),
-  down: queryInterface => queryInterface.bulkDelete('Users', null, {}),
+  down: (queryInterface) => queryInterface.bulkDelete('Users', null, {}),
```
can we remove the parentheses around queryInterface

chibytez commented on src/database/seeders/20190817154557-User.js:
```diff
@@ -2,7 +2,7 @@
 
 export default {
-  up: queryInterface => queryInterface.bulkInsert(
+  up: (queryInterface) => queryInterface.bulkInsert(
```
also, remove the parentheses around queryInterface

chibytez commented on src/controllers/logout.js:
```diff
@@ -0,0 +1,39 @@
+ * @class Logout
+ */
+
+class Logout {
```
can we reduce the line space between your JSDoc and your class, and make the jsDoc look like this
/**
 * @description class description
 * @class className
 */

davidfrank96 committed:
feat(eslint)
- ran the lint script to fix eslint
- added statuscode to the logout controller
[Finishes #167729790]

davidfrank96 committed:
feat(houndci)
- removed the parentises from the seeders and the migration files
[Finishes #167729790]

davidfrank96 committed:
feat(JsDocs)
- rearranged the JsDocs files in the logout file in the controller
[Finishes #167729790]

davidfrank96 committed:
feat(JsDocs)
- made changes to JsDocs
[Finishes #167729790]

davidfrank96 committed:
feat(HoundCI)
- made changes to logout.js file
[Finishes #167729790]

davidfrank96 committed:
feat(bugFix)
- made changes to logout.js file, index.js and logout.route.js
[Finishes #167729790]

davidfrank96 committed:
feat(codecovearge)
- Insatlled nyc for test and code coverage
[Finishes #167729790]

okabamac commented on src/controllers/logout.js:
```diff
@@ -0,0 +1,39 @@
+ */
+
+class Logout {
+  /**
```
Try to follow the jsDoc Chibuike provided below. That will help

okabamac commented on src/controllers/logout.js:
```diff
@@ -0,0 +1,30 @@
+ * @description Performs logout related related tasks
+ * @exports Logout
+ */
+class Logout {
```
Did you use this class?

davidfrank96 replied:
please can you elaborate on that pls

okafor-chidimma reviewed (approved):
Just pulled your branch and tested...all test cases passed...I approve for merge

okafor-chidimma reviewed (changes requested):
edit your eslint.json not to allow arrow-parens

okafor-chidimma reviewed (approved):
Looks good and I have tested it locally. Passed all test

chibytez reviewed (approved):
looks good

davidfrank96 committed:
feat(testfiles)
- reverted the change on the testfiles
[Finishes #167729790]

davidfrank96 committed:
feat(eslint)
- added some rules to eslint.js
[Finishes #167729790]

Stylll commented on src/index.js:
```diff
@@ -31,22 +32,25 @@ app.use(methodOverride());
-
 app.use(apiRoutes);
 
+app.use(session({
```
Why do we have this repeated here?

Stylll commented on src/routes/api/index.js:
```diff
@@ -1,31 +1,42 @@
+// });
+
+
+// export default usersRouter;
```
I see you commented these lines out. If they are not useful, you should take them out?. I noticed @chibytez and @okafor-chidimma also approved this PR. Do you think the commented lines should be left in the codebase?

davidfrank96 replied:
it will be taken off... I think this code is redundant

Stylll commented on src/test/app.test.js:
```diff
@@ -16,3 +16,27 @@ describe('TESTING ENDPOINTS', () => {
+      });
+  });
+
+  it('should return "you have logged out successfully" Page', (done) => {
```
You have duplicate test cases. Please remove one.

Stylll reviewed (changes requested):
@davidfrank96 Left a few comments. Please check them out. cc: @chibytez @okafor-chidimma

Stylll commented on src/test/app.test.js:
```diff
@@ -16,3 +16,16 @@ describe('TESTING ENDPOINTS', () => {
   });
 });
+
+describe('TESTING LOGOUT', () => {
```
we should move this test suite to its own file

davidfrank96 replied:
okay will do now...

Stylll reviewed (changes requested):
@davidfrank96 I dropped a comment. Kindly review and also fix the merge conflicts.

Stylll commented on src/routes/api/logout.route.js:
```diff
@@ -0,0 +1,10 @@
+const router = Router();
+const { logout } = Logout;
+
+router.get('/v1/auth/logout', logout);
```
I just noticed the logout route is a `GET` request. Is there a reason why this is not a POST request?.
Also, I see you named your route file `logout.route.js`, every route file in the codebase doesn't have the `**.route.js`. Please update your code to follow the existing pattern.

davidfrank96 replied:
on the passport docs it was a get request that's why i followed suit

Stylll reviewed (changes requested):
@davidfrank96 new changes request. Please review.

davidfrank96 committed:
feat(Feedback)
- added space to my travis.yml file
- deleted the redundant code in my index.js file
- removed the duplicate test case in the test file
- removed the duplicate session in the index.js file
[Finishes #167729790]

davidfrank96 committed:
feat(refactor)
- refactored the test scripts
- fix hound error on my test script
[Finishes #167729790]

Stylll commented on src/index.js:
```diff
@@ -7,6 +7,7 @@ import cors from 'cors';
 import errorhandler from 'errorhandler';
 import morgan from 'morgan';
 import methodOverride from 'method-override';
+
```
remove empty space

Stylll commented on src/index.js:
```diff
@@ -34,6 +35,7 @@ app.use(express.static(`${__dirname}/public`));
 app.use(passport.initialize());
 app.use(passport.session());
 
+
```
remove empty space

Stylll commented on src/routes/api/index.js:
```diff
@@ -1,11 +1,15 @@
-
+// import { Router } from 'express';
```
remove the commented line

Stylll commented on src/routes/api/index.js:
```diff
@@ -1,11 +1,15 @@
 usersRouter.use('/', socialMediaRoutes);
 
+// const usersRouter = Router();
+
```
remove the commented line and empty space

Stylll reviewed (changes requested):
Please pay attention and ensure to review your code properly. There are a few empty spaces and commented lines. I dropped comments on them, kindly remove them.

davidfrank96 committed:
feat(parentesis)
-removed the parentesis on my seeders file
[Finishes #167729790]

davidfrank96 committed:
feat(mergeConflicts)
- Resolved merged conflicts
[Finishes #167729790]

davidfrank96 committed:
feat(Houndci)
- solving houndci issues
[Finishes #167729790]

davidfrank96 committed:
feat(Houndci)
-code refactor to houndci
[Finishes #167729790]

davidfrank96 committed:
feat(Houndci)
-code refactor to houndci to the test files
[Finishes #167729790]

davidfrank96 committed:
feat(Houndci)
-code refactor to houndci
[Finishes #167729790]

davidfrank96 committed:
feat(Houndci)
- ran  the lint script
[Finishes #167729790]

davidfrank96 committed:
feat(testfile)
-used a different file for my test cases
[Finishes #167729790]

davidfrank96 committed:
feat(feadback)
- implemented feedback
[Finishes #167729790]

davidfrank96 committed:
feat(mergeConflicts)
-resolving merge conflicts
[Finishes #167729790]

davidfrank96 committed:
feat(feedback)
-removed all the variables from the .env.emaple file
-adjusted the logout.route file to logout.js
[Finishes #167729790]

davidfrank96 committed:
feat(feedback)
- changing the GET request to POST request
[Finishes #167729790]

davidfrank96 committed:
feat(feedbacks)
-made some feebacks
-Removed wide spaces
[Finishes #167729790]

davidfrank96 committed:
feat(MergeConflicts)
- resolving merge conflicts
[Finishes #167729790]

davidfrank96 committed:
feat(spaces)
-removing sapces
[Finishes#167729790]
